### ISSUE TITLE
Add foreign key sample code to example project.

### ISSUE THREAD

**Issue by curioustechizen** [mitmel/SimpleContentProvider#10]
The current example does a great job of demonstrating how to use `ContentItem` and `SimpleContentProvider`. It would be great to add an example of using foreign keys in there.

**Comment by xxv** [comment]
Agreed. There is currently unit test code that does it, but a proper example would be good. 

Relatedly, the M2M system is being massively overhauled so that it's easier to use. That'll need an example as well.
